common/lib/dependabot/pull_request_creator/branch_namer/solo_strategy.rb

```diff
@@ -221,6 +232,30 @@ def dependency_digest
             T.nilable(String)
           )
         end
+
+        sig do
+          params(requirements: T.nilable(T::Array[T::Hash[Symbol, T.untyped]]))
+            .returns([T.nilable(String), T.nilable(String)])
+        end
+        def extract_tag_and_digest(requirements)
+          return [nil, nil] unless requirements
+
+          tag = requirements.filter_map { |r| r.dig(:source, "tag") || r.dig(:source, :tag) }.first
+          digest = requirements.filter_map { |r| r.dig(:source, "digest") || r.dig(:source, :digest) }.first
+          [tag, digest]
+        end
+
+        sig { params(tag: T.nilable(String), digest: T.nilable(String)).returns(T.nilable(String)) }
+        def format_digest_branch_name(tag, digest)
+          return tag if digest.nil?
+
+          digest_parts = digest.split(":")
+          if tag && !tag.empty?
+            "#{tag}-#{digest_parts.first}#{digest_parts.last&.slice(0, 6)}"
+          else
+            "#{digest_parts.first}#{digest_parts.last&.slice(0, 6)}"
```

Comment:
Can you add the separator back here? Or if `:` isn't allowed in branch names, use `-`?

Reply:
Added -